dialog/dialog/relational-algebra.md

```diff
@@ -0,0 +1,415 @@
+
+For example:
+
+```
+users = [
+    (user_id: 1, user_name: "Alice"),
+    (user_id: 2, user_name: "Bob"),
+    (user_id: 3, user_name: "Mallory")
+]
+
+admins = [
+    (user_id: 1),
+    (user_id: 2),
+]
+
+semijoin(users, admins) [
+    (user_id: 1, user_name: "Alice"),
+    (user_id: 2, user_name: "Bob"),
+]
+```
+
+## 3.11 Antijoin
+
+An antijoin is a binary operator which returns all tuples in its first input relation such that there is no tuple in its second relation for which the common attributes of both tuples are equal.
+
+This operation is equivalent to taking the [difference](#35-difference) of the first relation and the [semijoin](#310-semijoin) of both, however implementations MAY implement antijoins as a specialization of these operations, for performance reasons.
+
+## 3.12 Group By
+
+Group by is the mechanism by which aggregation is performed. Group by is performed over a relation with respect to a set of grouping attributes, and an aggregate function. The operation returns the set of tuples computed by first grouping the input relation, and then applying the aggregate function to each group.
+
+Implementations MAY support user defined aggregates, but MUST support the aggregate functions described in the [specification for the query language](query-engine.md#aggregation).
+
+For example:
+
+```
+products = [
+    (product_id: 1, product_category_id: 1, product_stock: 3),
+    (product_id: 2, product_category_id: 1, product_stock: 5),
+    (product_id: 3, product_category_id: 2, product_stock: 12),
+    (product_id: 4, product_category_id: 2, product_stock: 7)
+]
+
+# TODO Not a fan of this notation: returning a pair of the group and a scalar doesn't fit with the rest of the specification.
+#      In my prototype, aggregation is always over a variable, and the result is named using that variable. That's not
+#      how relational algebra is usually presented though, so I should revisit this as I describe more of what the overall
+#      compilation looks like.
+group_by(product_category_id, sum(product_stock), products) => [
+    (product_category_id: 1, 8),
+    (product_category_id: 2, 19),
+]
+```
+
+# 4. Propositional Formula
+
+Selection criterion for operations like [selection](#33-selection) and [theta joins](#38-theta-join) are specified in terms of a small subset of propositional logic.
+
+A propositional formula is canonically represented in disjunctive normal form, as a disjunction of conjunctives:
+
+```
+(A1 and A2 and ... and An) or ... or (Z1 and Z2 and ... Zn)
+```
+
+Each conjunctive term specifies a binary relation between attributes and constants, using the relational operators: `<`, `<=`, `=`, `!=`, `>=`, `>`, and these relations MAY also be negated.
```

Comment:
> Is it always an attribute on the left and a constant on the right?

Nope, it can be either type on either side. Thanks, I'll make that less ambiguous!